source/extensions/filters/network/thrift_proxy/app_exception_impl.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "extensions/filters/network/thrift_proxy/filters/filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+
+/**
+ * Thrift Application Exception types.
+ * See https://github.com/apache/thrift/blob/master/doc/specs/thrift-rpc.md
+ */
+enum class AppExceptionType {
+  Unknown = 0,
+  UnknownMethod = 1,
+  InvalidMessageType = 2,
+  WrongMethodName = 3,
+  BadSequenceId = 4,
+  MissingResult = 5,
+  InternalError = 6,
+  ProtocolError = 7,
+  InvalidTransform = 8,
+  InvalidProtocol = 9,
+  UnsupportedClientType = 10,
```

Comment:
`fbthrift` has a few more exception types than `apache thrift`: https://github.com/facebook/fbthrift/blob/04ba53d4503d51a9a27f700b114eb7cea4678dc1/thrift/lib/cpp/TApplicationException.h#L57. Should these be supported too? As an example later in the code `InternalError` could be replaced by more specific code such as `LoadShedding`. Would that create sub-optimal experience for `apache thrift` as the enum value would not exist in the client services logic? Without using the `fbthrift` exception types there will be feature regressions for some client services as would no longer be able to differentiate.

Is `apache thrift` being treated as the standard here when there are competing standards?

Reply:
I think short-term, we'll stick with the apache definitions. Longer term, we'll probably want to use internal identifiers for various error conditions and then provide a configuration point to control whether they get mapped to apache or fb's values.